### PR TITLE
ci: travis scaffolding

### PR DESCRIPTION
**Pull request by flyinprogrammer**

### CODE REVIEW

File: ci/travis.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+echo -e "TRAVIS_BRANCH=$TRAVIS_BRANCH"
+echo -e "TRAVIS_TAG=$TRAVIS_TAG"
+echo -e "TRAVIS_COMMIT=${TRAVIS_COMMIT::7}"
+echo -e "TRAVIS_PULL_REQUEST=$TRAVIS_PULL_REQUEST"
+
+if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then
+    # Pull Request
+    echo -e "Build Pull Request #$TRAVIS_PULL_REQUEST => Branch [$TRAVIS_BRANCH]"
+    ./mvnw
```
Comment (flyinprogrammer) [inline ci/travis.sh]: make these do what ever lifecycle pieces you want

Comment (OlegDokuka) [inline ci/travis.sh]:
```suggestion
    ./mvnw clean install
```

File: ci/travis.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+elif [ "$TRAVIS_PULL_REQUEST" == "false" ] && [ "$TRAVIS_BRANCH" == "develop" ] && [ "$TRAVIS_TAG" == "" ]; then
+    # Develop Branch
+    echo -e 'Build Branch with Snapshot => Branch ['$TRAVIS_BRANCH']'
+    export ORG_GRADLE_PROJECT_releaseType=snapshot
+    ./mvnw
```
Comment (OlegDokuka) [inline ci/travis.sh]:
```suggestion
    ./mvnw clean install publish -Drevision-prefix=BUILD-SNAPSHOT
```

File: ci/travis.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+elif [ "$TRAVIS_PULL_REQUEST" == "false" ] && [[ "$TRAVIS_BRANCH" == release/* ]] && [ "$TRAVIS_TAG" == "" ]; then
+    # Release Branch
+    echo -e 'Build Branch for Release => Branch ['$TRAVIS_BRANCH']'
+    ./mvnw
```
Comment (OlegDokuka) [inline ci/travis.sh]:
```suggestion
    ./mvnw clean install publish -Drevision-prefix=RC
```

File: ci/travis.sh [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+elif [ "$TRAVIS_PULL_REQUEST" == "false" ] && [ "$TRAVIS_BRANCH" == "master" ]; then
+    # Master Branch
+    echo -e 'Build Master for Release => Branch ['$TRAVIS_BRANCH']'
+    export ORG_GRADLE_PROJECT_releaseType=release
+    ./mvnw
```
Comment (OlegDokuka) [inline ci/travis.sh]:
```suggestion
    ./mvnw clean install
```

File: ci/travis.sh [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+elif [ "$TRAVIS_PULL_REQUEST" == "false" ] && [ "$TRAVIS_TAG" != "" ]; then
+    # Tag
+    echo -e 'Build Tag for Release => Tag ['$TRAVIS_TAG']'
+    export ORG_GRADLE_PROJECT_releaseType=release
+    ./mvnw
```
Comment (OlegDokuka) [inline ci/travis.sh]:
```suggestion
    ./mvnw clean install publish -Drevision-prefix=RELEASE
```

File: ci/travis.sh [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+else
+    # Feature Branch
+    echo -e 'Build Branch => Branch ['$TRAVIS_BRANCH']'
+    export ORG_GRADLE_PROJECT_releaseType=snapshot
+    ./mvnw
```
Comment (OlegDokuka) [inline ci/travis.sh]:
```suggestion
    ./mvnw clean install publish -Drevision-prefix=BUILD-SNAPSHOT
```

File: ci/travis.sh [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+echo -e "TRAVIS_PULL_REQUEST=$TRAVIS_PULL_REQUEST"
+
+if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then
+    # Pull Request
+    echo -e "Build Pull Request #$TRAVIS_PULL_REQUEST => Branch [$TRAVIS_BRANCH]"
+    ./mvnw
+elif [ "$TRAVIS_PULL_REQUEST" == "false" ] && [ "$TRAVIS_BRANCH" == "develop" ] && [ "$TRAVIS_TAG" == "" ]; then
+    # Develop Branch
+    echo -e 'Build Branch with Snapshot => Branch ['$TRAVIS_BRANCH']'
+    export ORG_GRADLE_PROJECT_releaseType=snapshot
```
Comment (OlegDokuka) [inline ci/travis.sh]:
```suggestion
```

File: ci/travis.sh [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    ./mvnw
+elif [ "$TRAVIS_PULL_REQUEST" == "false" ] && [[ "$TRAVIS_BRANCH" == release/* ]] && [ "$TRAVIS_TAG" == "" ]; then
+    # Release Branch
+    echo -e 'Build Branch for Release => Branch ['$TRAVIS_BRANCH']'
+    ./mvnw
+elif [ "$TRAVIS_PULL_REQUEST" == "false" ] && [ "$TRAVIS_BRANCH" == "master" ]; then
+    # Master Branch
+    echo -e 'Build Master for Release => Branch ['$TRAVIS_BRANCH']'
+    export ORG_GRADLE_PROJECT_releaseType=release
```
Comment (OlegDokuka) [inline ci/travis.sh]:
```suggestion
```

File: ci/travis.sh [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    ./mvnw
+elif [ "$TRAVIS_PULL_REQUEST" == "false" ] && [ "$TRAVIS_TAG" != "" ]; then
+    # Tag
+    echo -e 'Build Tag for Release => Tag ['$TRAVIS_TAG']'
+    export ORG_GRADLE_PROJECT_releaseType=release
```
Comment (OlegDokuka) [inline ci/travis.sh]:
```suggestion
```

File: ci/travis.sh [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+    ./mvnw
+else
+    # Feature Branch
+    echo -e 'Build Branch => Branch ['$TRAVIS_BRANCH']'
+    export ORG_GRADLE_PROJECT_releaseType=snapshot
```
Comment (OlegDokuka) [inline ci/travis.sh]:
```suggestion
```

**Review by OlegDokuka — CHANGES_REQUESTED** [review]

**Pull request merged**